src/internal/git/utils.go

```diff
@@ -232,3 +240,95 @@ func deleteBranchIfExists(gitDirectory string, branchName plumbing.ReferenceName
 
 	return nil
 }
+
+func CreateReadOnlyUser() error {
```

Comment:
nit but this whole function could use some readability work, maybe a few more comments about what the REST calls or doing or even some line breaks added.  Realize go isn't doing any favors here, but it was a lot to read through bunched together--especially on github:-).  It might be more readable (and DRY) to wrap some of the REST call boilerplate.

Reply:
@jeff-mccoy This should be more readable now